packages/compiler/src/extractor/extract.ts

```diff
@@ -41,12 +41,14 @@ export const extractProps = (
     ...(variant?.baseStyle as Record<string, string>),
   };
 
-  const defaultProps = componentDefaultProps(componentName);
+  const systemDefaultProps = componentDefaultProps(componentName);
+  const userDefaultProps = variant?.defaultProps;
```

Comment:
I had previously declared a variable named `componentVariantProps` to accumulate variant-related props. However, you introduced a new variable `userDefaultProps` and did not utilize `componentVariantProps`, which I believe is perfectly fine. I would kindly suggest renaming `componentVariantProps` to something more descriptive like `baseStyle` to better align with its intended use case.